Description: convert the mimetype definitions to the XDG format
             also remove the old KDE 3 definition
Author: Pino Toscano <pino@kde.org>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,8 @@ PROJECT( kile )
 find_package(KDE4 4.2.0 REQUIRED)
 include (KDE4Defaults)
 
+find_package( SharedMimeInfo REQUIRED )
+
 
 SET( KILE_VERSION CACHE STRING "Kile's version; can be used to install special documentation files in share/doc/kile-$KILE_VERSION/" )
 
@@ -131,7 +133,8 @@ ADD_SUBDIRECTORY( src )
 
 ########### install files ###############
 
-INSTALL( FILES x-kilepr.desktop DESTINATION share/mimelink/text )
+install( FILES kile.xml DESTINATION ${XDG_MIME_INSTALL_DIR} )
+update_xdg_mimetypes( ${XDG_MIME_INSTALL_DIR} )
 
 INSTALL( FILES kile.upd kile1.6_upd.pl kile1.7_upd.pl kile1.9_upd.pl
 kile1.9-1_upd.pl kile2.0-1_upd.pl kile2.0-2_upd.pl kile2.0-3_upd.pl
--- /dev/null
+++ b/kile.xml
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<mime-info xmlns="http://www.freedesktop.org/standards/shared-mime-info">
+  <mime-type type="text/x-kilepr">
+    <comment>Kile project</comment>
+    <glob pattern="*.kilepr"/>
+  </mime-type>
+</mime-info>
--- a/x-kilepr.desktop
+++ /dev/null
@@ -1,63 +0,0 @@
-[Desktop Entry]
-Encoding=UTF-8
-Comment=Kile Project
-Comment[ar]=مشروع Kile
-Comment[ast]=Proyeutu Kile
-Comment[be]=Праект Kile
-Comment[bg]=Проект на Kile
-Comment[ca]=Projecte Kile
-Comment[ca@valencia]=Projecte Kile
-Comment[cs]=Kile projekt
-Comment[csb]=Projekt Kile
-Comment[da]=Kile-projektet
-Comment[de]=Kile-Projekt
-Comment[el]=Έργο Kile
-Comment[en_GB]=Kile Project
-Comment[eo]=Kile Projekto
-Comment[es]=Proyecto Kile
-Comment[et]=Kile Projekt
-Comment[fi]=Kile-projekti
-Comment[fr]=Projet Kile
-Comment[ga]=Tionscadal Kile
-Comment[gl]=O proxecto Kile
-Comment[hi]=काइल परियोजना
-Comment[hne]=काइल परियोजना
-Comment[hr]=Projekt Kile
-Comment[hu]=Kile Projekt
-Comment[it]=Progetto di Kile
-Comment[ja]=Kile プロジェクト
-Comment[kk]=Kile жобасы
-Comment[km]=គម្រោង Kile
-Comment[ko]=Kile 프로젝트
-Comment[lt]=Kile projektas
-Comment[lv]=Kile projekts
-Comment[ml]=കൈല്‍ സംരംഭം
-Comment[ms]=Projek Kile
-Comment[nb]=Kile-prosjektet
-Comment[nds]=Kile-Projekt
-Comment[ne]=किल परियोजना
-Comment[nl]=Kile-project
-Comment[nn]=Kile-prosjekt
-Comment[pa]=ਕਿਲੀ ਪਰੋਜੈਕਟ
-Comment[pl]=Projekt Kile
-Comment[pt]=Projecto do Kile
-Comment[pt_BR]=Projeto Kile
-Comment[ro]=Proiect Kile
-Comment[ru]=Проект Kile
-Comment[sk]=Kile projekt
-Comment[sv]=Kile-projekt
-Comment[th]=Kile Project
-Comment[tr]=Kile Projesi
-Comment[uk]=Проект Kile
-Comment[wa]=Pordjet Kile
-Comment[x-test]=xxKile Projectxx
-Comment[zh_CN]=Kile 工程
-Comment[zh_TW]=Kile 專案
-Icon=kile
-MimeType=text/x-kilepr
-Patterns=*.kilepr;
-Type=MimeType
-
-[Property::X-KDE-text]
-Type=bool
-Value=true
